Dockerfile

```diff
@@ -344,7 +344,7 @@ ARG TAG
 ARG VERSION_PKG="github.com/talos-systems/talos/pkg/version"
 RUN --mount=type=cache,target=/.cache/go-build GOOS=linux GOARCH=amd64 go test -c \
```

Comment:
should this be go build?

Reply:
we can't use `go build` afaik to build unit-tests, the trick is `go test -c` compiles the test binary, but doesn't run it. so we can export that binary back and use it as a test runner every time we need to have an integration test